basic-javascript/tests/project-4.test.js

```diff
@@ -1,4 +1,231 @@
 const assert = require('chai').assert;
+let sinon = require("sinon");
 const funcs = require('../src/project-4.js');
 
-// whoops.. there is no test suite for this file. You'll simply just have to create one :/
\ No newline at end of file
+// whoops.. there is no test suite for this file. You'll simply just have to create one :/
+
+describe('Project-4 Functions', () => {
+
+  describe('multiplyArguments', () => {
+
+    it('should be a function', () => {
+      const multiplyArguments = funcs.multiplyArguments;
+      assert.isFunction(multiplyArguments);
+    });
+    it('should return 0 is no arguments are passed in', () => {
+      const multiplyArguments = funcs.multiplyArguments;
+      const result = multiplyArguments();
+      assert.equal(result, 0);
+    });
+
+    it('should correctly multiply the arguments', () => {
+      const multiplyArguments = funcs.multiplyArguments;
+      const result = multiplyArguments(5,4,5);
+      assert.equal(result, 100);
+    });
+  });
+
+  describe('invokeCallback', () => {
+    it('should be a function', () => {
+      const invokeCallback = funcs.invokeCallback;
+      assert.isFunction(invokeCallback);
+    });
+    it('should invoke callback', () => {
+      const invokeCallback = funcs.invokeCallback;
+      const callback = sinon.spy()
+      const result = invokeCallback(callback);
+      sinon.assert.calledOnce(callback);
+    });
+  });
+
+  describe('sumArray', () => {
+    it('should be a function', () => {
+      const sumArray = funcs.sumArray;
+      assert.isFunction(sumArray);
+    });
+    it('should invoke callback', () => {
+      const sumArray = funcs.sumArray;
+      const numbers = [1, 2, 3];
+      const callback = sinon.spy();
+      const result = sumArray(numbers, callback);
+      sinon.assert.calledOnce(callback);
+    });
+    it('should correctly sum the array', () => {
+      const sumArray = funcs.sumArray;
+      const numbers = [1, 2, 3];
+      const callback = sinon.spy();
```

Comment:
Here, you can pass the `sumArray` function to `sinon.spy` in order to have it emulate the functionality of that callback. That way you can still test the actual result of the callback, along with being able to check all of the properties that `sinon.spy` allows you to check.